File: 01-frontend/angular-game-ranking-system/src/app/common/clan.ts
import {User} from "./user";

export class Clan {

  public id: number;
  public name: string;
  public maxNumOfMembers: number;
  public requiredTrophies: number;
  public totalMembers: number;
  public totalTrophies: number;

  constructor(

  ) {
    this.id = 0;
    this.name = '';
    this.maxNumOfMembers = 0;
    this.requiredTrophies = 0;
    this.totalMembers = 0;
    this.totalTrophies = 0;
  }
}

File: 02-backend/spring-boot-game-ranking-system/src/main/java/com/sithuhantun/game_ranking_system/model/Clan.java
package com.sithuhantun.game_ranking_system.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "clan")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
public class Clan {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "max_num_of_members")
    private int maxNumOfMembers;

    @Column(name = "required_trophies")
    private int requiredTrophies;

    @OneToMany(mappedBy = "clan",
            fetch = FetchType.EAGER,
            cascade = {CascadeType.PERSIST, CascadeType.MERGE,
            CascadeType.REFRESH, CascadeType.DETACH})
    @JsonIgnore
    private List<User> playerList;

    public Clan(String name, int maxNumOfMembers, int requiredTrophies) {
        this.name = name;
        this.maxNumOfMembers = maxNumOfMembers;
        this.requiredTrophies = requiredTrophies;
    }
}
